File: 8_kyu/Multiplication_table_for_number.go
package kata

import (
	"bytes"
	"fmt"
)

func MultiTable(number int) string {
	var accu bytes.Buffer
	for i := 1; i < 11; i++ {
		fmt.Fprintf(&accu, "%d * %d = %d\n", i, number, i*number)
	}
	// remove the trailing \n. It has 1 byte length.
	accu.Truncate(accu.Len() - 1)
	return accu.String()
}

File: 8_kyu/Multiplication_table_for_number.rs
use std::fmt::Write;

fn multi_table(n: u64) -> String {
    let mut result = String::new();
    for i in 1..=10 {
        writeln!( result, "{} * {} = {}", i, n, i*n );
    }
    // remove the last \n
    result.pop();
    result
}